pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -33,145 +33,72 @@ type StorageCohort struct {
-
-	// DataProtectionCapabilities list data protection capabilities in case of failures
-	DataProtectionCapabilities StorageCohortDataProtectionCapabilities `json:"dataProtectionCapabilities,omitempty"`
-}
-
-// StorageCohortDataStorageCapabilities
-type StorageCohortDataStorageCapabilities struct {
-	// Contains the types of access modes supported by the cohort
-	AccessModes []StorageCohortAccessMode `json:"accessModes,omitempty"`
-
-	// ProvisioningPolicies defines what type of pool/volume can be created inside the cohort.
-	ProvisioningPolicies []ProvisioningPolicy `json:"provisioning_policies,omitempty"`
-
-	// Multipathing describes the different configurations of accessing
-	// storage through different nodes within a cohort
-	Multipathing StorageCohortMultipathingOption `json:"multipathing,omitempty"`
-
-	// If specified, the compression algorithm supported by the cohort
-	// +nullable
-	Compression *[]StorageCohortCompressionAlgorithm `json:"compression,omitempty"`
-
-	// Deduplication, if set to true, ensures that the data is de-duplicated within the cohort
-	// +nullable
-	Deduplication *bool `json:"deduplication,omitempty"`
-}
-
-// StorageCohortDataSecurityCapabilities
-type StorageCohortDataSecurityCapabilities struct {
-	// MediaEncryption either specifies whether encryption is supported or specifies the encryption algorithm itself
-	// +nullable
-	MediaEncryption *[]StorageCohortEncryptionAlgorithm `json:"mediaEncryption,omitempty"`
-
-	// DataSanitizationPolicy specifies the supported data sanity checks
-	DataSanitizationPolicy StorageCohortDataSanitizationPolicy `json:"dataSanitizationPolicy,omitempty"`
-}
-
-// StorageCohortIOConnectivityCapabilities
-type StorageCohortIOConnectivityCapabilities struct {
-	// Contains the list of access protocols supported by the cohort
-	AccessProtocols []StorageCohortAccessProtocol `json:"accessProtocols,omitempty"`
-}
-
-// StorageCohortIOPerformanceCapabilities
-type StorageCohortIOPerformanceCapabilities struct {
-	// Average I/O operation latency(in microseconds) allowed within the cohort
-	AverageIOOperationLatencyMicroseconds uint64 `json:"averageIoOperationLatencyMicroseconds,omitempty"`
-
-	// Max I/O operation(per second/TB) allowed within the cohort
-	MaxIOOperationsPerSecondPerTerabyte uint64 `json:"maxIoOperationsPerSecondPerTerabyte,omitempty"`
-
-	// List of storage tiers supported by the cohort
-	StorageTier []StorageCohortStorageTier `json:"storageTier,omitempty"`
-}
-
-// StorageCohortDataProtectionCapabilities
-type StorageCohortDataProtectionCapabilities struct {
-	// Whether replication is supported by the cohort
-	// +optional
-	Replication *bool `json:"replication,omitempty"`
-
-	// Backup controls data can be backed up by the cohort
 	// +optional
-	Backup *bool `json:"backup,omitempty"`
-
-	// whether cohort will support snapshots
-	// +optional
-	Snapshots *bool `json:"snapshots,omitempty"`
+	Capabilities Capabilities `json:"capabilities,omitempty"`
```

Comment:
How is this optional?

Reply:
Yes in a way. This represent the aggregate capabilities of all the pools which are updated by pool provisioners after pool provisioning/deprovisioning. This also helps volume scheduler to narrow down on the pool selection for a volume.